packages/js/src/utils/id-token.ts

```diff
@@ -0,0 +1,39 @@
+import { UrlSafeBase64 } from '@silverhand/essentials';
+import * as s from 'superstruct';
+
+/**
+ * @link [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)
+ */
+const IdTokenClaimsSchema = s.type({
+  iss: s.string(),
+  sub: s.string(),
+  aud: s.string(),
+  exp: s.number(),
+  iat: s.number(),
+  at_hash: s.optional(s.string()),
+});
+
+export type IdTokenClaims = s.Infer<typeof IdTokenClaimsSchema>;
+
+export const decodeIdToken = (token: string): IdTokenClaims => {
+  const { 1: encodedPayload } = token.split('.');
+
+  if (!encodedPayload) {
+    throw new Error('invalid token');
+  }
+
+  const json = UrlSafeBase64.decode(encodedPayload);
+
+  try {
+    // Use Superstruct to validate the json type
+    const idToken = JSON.parse(json) as IdTokenClaims;
+    s.assert(idToken, IdTokenClaimsSchema);
+    return idToken;
+  } catch (error: unknown) {
+    if (error instanceof s.StructError) {
+      throw error;
+    }
+
+    throw new Error('invalid token: JSON parse failed');
+  }
```

Comment:
no need for try - catch 😂

Reply:
Removed, PTAL.

Reply:
This is based on the previous comments. The try..catch might be valid if we have to keep JSON.parse and throw a controllable error type.

Reply:
@wangsijie Any idea?

Reply:
Conclusion: No need for try-catch ( See [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1642588712357800) )